uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/PowerLiftIncidentRule.java

```diff
@@ -31,21 +32,33 @@ public void evaluate() throws Throwable {
                 try {
                     base.evaluate();
                 } catch (final Throwable originalThrowable) {
+                    String powerLiftIncidentDetails = null;
                     try {
                         Logger.e(
                                 TAG,
                                 "Encountered error during test....creating PowerLift incident.",
                                 originalThrowable
                         );
-                        powerLiftIntegratedApp.createPowerLiftIncident();
+                        powerLiftIncidentDetails = powerLiftIntegratedApp.createPowerLiftIncident();
                     } catch (final Throwable powerLiftError) {
                         Logger.e(
                                 TAG,
                                 "Oops...something went wrong...unable to create PowerLift incident.",
                                 powerLiftError
                         );
                     }
-                    throw originalThrowable;
+                    if (TextUtils.isEmpty(powerLiftIncidentDetails)) {
+                        throw originalThrowable;
+                    } else {
+                        final String message = originalThrowable.getMessage() + "\n" +
+                                "PowerLift Incident Created via " +
+                                ((App) powerLiftIntegratedApp).getAppName() +
+                                " - " + powerLiftIncidentDetails.trim();
+                        throw new Throwable(
```

Comment:
Should we actually give this a type?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1211/commits/62609846c5df0ead8b45fe6e2521df82af99bea5